node/network/approval-distribution/src/lib.rs

```diff
@@ -500,12 +524,14 @@ impl State {
 		// split_off returns everything after the given key, including the key
 		let split_point = finalized_number.saturating_add(1);
 		let mut old_blocks = self.blocks_by_number.split_off(&split_point);
+
 		// after split_off old_blocks actually contains new blocks, we need to swap
 		std::mem::swap(&mut self.blocks_by_number, &mut old_blocks);
 
 		// now that we pruned `self.blocks_by_number`, let's clean up `self.blocks` too
-		old_blocks.values().flatten().for_each(|h| {
-			self.blocks.remove(h);
+		old_blocks.values().flatten().for_each(|relay_block| {
+			self.recent_finalized_blocks.note_finalized(relay_block);
```

Comment:
point of information: note that it's not just finalized blocks that are pruned but also all competing blocks. the '10' buffer approach is still ~ok but it's much more likely to get overturned, especially if there are longish jumps in finality

Reply:
Increased to 20 and renamed to `_outdated/Outdated`